Mantis 44891: Bugs and usability issues when copying role permissions

ingoj opened the pull request:
possible fixes for mantis 44891

buttons which start copying of permissions are relabeled
on-screen-message additionally for selecting target roles displays source role
copy behaviour form has a new on-screen-message that shows copy source and up to three targets (if more targets are selected, the number of selected targets is displayed)

this fix introduces 4 new language variables, which are provided in german and english, all in rbac:
   rbac_copy_role_behaviour_info
   rbac_copy_multi_targets
   rbac_copy_no_targets
   rbac_select_roles

ingoj committed:
Fix Usability of RoleCopy Workflow

Add Infos about copy source and targets (up to three targets are sent literally, four and mor targets present number of targets) in messageboxes
Fix if no target role is selectes
Fix copyright notice position in code

ingoj committed:
Fix usability of  roles copy

adjust langvar
eliminate dysfunctional filter on target role selection screen

ingoj committed:
switch lagvar in copy roles

ingoj committed:
fix typo and add typecast

ingoj committed:
adjust langvars de for usability fixes copy roles

new vars for rbac:
rbac_copy_behaviour_info
rbac_copy_multi_targets
rbac_copy_no_targets
rbac_select_roles

ingoj committed:
adjust language for roles copy fixes

new variables in rbac:

rbac_copy_behaviour_info
rbac_copy_multi_targets
rbac_copy_no_targets
rbac_select_roles

ingoj committed:
fix typo

ingoj committed:
fix typo

ingoj commented:
in class.ilObjRoleFolderGUI copyright notice is adjusted, too

ingoj commented:
Probably this PR needs to be discussed in JF

kergomard reviewed (changes requested):
Thank you very much @ingoj for the PR!

There are a few little things concerning code-styles:
- [ ] Could you please fix the code-style violations (use `php ./libs/composer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix --config=./CI/PHP-CS-Fixer/code-format.php_cs --using-cache=no -vvv <path to file>` from the root of your ILIAS installation to do so).
- [ ] Please rename the variables you introduced from camelCase to snake_case (e.g. [here](https://github.com/ILIAS-eLearning/ILIAS/pull/9394/files#diff-e757573375fadb0c900fb0c008759c7cbb108fe9ac2ca81353590e059739d54dR299)).
- [ ] Please remove the variable `$sourceName` and move the corresponding retrieval-call to its single use.
- [ ] I don't like the `switch`-statement. I think it makes things very hard to read. Please create a function `buildTargetNamesString(array $targets): string` with an early-return structure with `if`s.
- [ ] Please copy the content of the language variable 'user_role_selection' in all languages to the variable 'rbac_select_roles'. That avoids that everybody needs to retranslate this value.

I hope I didn't miss anything and sorry if I'm not precise enough here.

I think we can do this without going to the JF.

Thanks again and best,
@kergomard

ingoj committed:
Merge branch 'ILIAS-eLearning:release_9' into release_9_rolecopy

ingoj committed:
Streamline code and additional source name in target search

addition: display first two selected targets and remaining count if more than three targets are selected

ingoj committed:
adjust language to changes in RoleFolderGUI

ingoj committed:
adjust language to changes in RoleFolderGUI

ingoj committed:
fix typo

$ -> %

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
U

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
up

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
up

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
Update for copy roles process

ingoj committed:
fix misspelled function

ingoj committed:
fix cs violations

ingoj commented:
I updated my repo according to your requested changes. Just as a remark: Only in 2 languages (beside english and german) translations were preset. Nevertheless I copied the variable for all languages, I hope this was right.

ingoj committed:
Update ilias_en.lang

add missing #

kergomard commented:
Thank you very much @ingoj for the PR!

To avoid more forth and back I took your changes, added a few of my own and committed them in https://github.com/ILIAS-eLearning/ILIAS/commit/e87cc1e01bde39458983d9dce511af5f8012f177 . I picked the changes to ILIAS 10 and trunk.

Thanks again and best,
@kergomard
